### PR TITLE
uses media_type for MIME type in webhooks

### PR DESCRIPTION
**Pull request by yoLotus**
## Description

As describe in this issue #179, Rails displays a warning when reading the *MIME* type with `content_type` from the `request` object. I suggest to prioritize the `media_type` when existing.

## CHANGELOG

* [CHANGED] Fix Rails 7.1 MIME type deprecation warning

### CODE REVIEW

**Comment by fbenevides** [comment]
Hi @yoLotus, thanks for the contribution.

It still needs a spec to cover both scenarios. Could you please add it, so we can merge it?

**Comment by yoLotus** [comment]
Hello @fbenevides , sorry for the late response and thanks you a lot for having taken a look.

I'm really not sure of my solution for the test, wondering if my mock it's acceptable or not. 

It seems that `ActionDispatch::Request` use the `Rack::Request` [under the hood](https://github.com/rails/rails/blob/04972d9b9ef60796dc8f0917817b5392d61fcf09/actionpack/lib/action_dispatch/http/request.rb#L60) so I use a delegator to be as close as possible of the Rails object. 
What do you think?

**Comment by sonologico** [comment]
I think we could simply replace `content_type` with `media_type`. `media_type` has been available for enough time that I don't think we need a fallback to `content_type`.

File: lib/pusher/webhook.rb [hunk 1]
```diff
@@ -31,10 +31,10 @@ class WebHook
     def initialize(request, client = Pusher)
       @client = client
       # For Rack::Request and ActionDispatch::Request
-      if request.respond_to?(:env) && request.respond_to?(:content_type)
+      if request.respond_to?(:env) && (request.methods & [:media_type, :content_type]).any?
         @key = request.env['HTTP_X_PUSHER_KEY']
         @signature = request.env["HTTP_X_PUSHER_SIGNATURE"]
-        @content_type = request.content_type
+        @content_type = request.media_type || request.content_type
```
Comment (yoLotus) [inline lib/pusher/webhook.rb:37]: @sonologico Just to be sure, do you propose to remove any occurrence of `content_type` in this snippet and just read  directly the `request.media_type`?

**Comment by r7kamura** [comment]
I too have encountered this problem and would be happy if this Pull Request is merge as soon as possible.

The reason for this Pull Request and the related Issue is to address the `ActionDispatch::Request#content_type` warning, but that warning is not the only problem.

In Rails apps that have been `rails new` since Rails 7.0 or have switched to `load_defaults 7.0`, the behavior of `ActionDispatch::Request#content_type` has already changed and it will return a value like `"application/json; charset=utf-8"`, so the following case-when condition no longer works as intended:

https://github.com/pusher/pusher-http-ruby/blob/18109ec781501c673fb1853869cf89f8ed27296d/lib/pusher/webhook.rb#L86-L91

This is also a problem if you are simply using Rack, not Rails, because it is also not working as intended since `Rack::Request#content_type` will return a value like `"Content-Type: application/json; charset=utf-8"` as well.

I think it would also be better to have a test case for `Content-Type: application/json; charset=utf-8`, which would fail before this change.

File: lib/pusher/webhook.rb (hunk 1 above)
Comment (sonologico) [reply lib/pusher/webhook.rb:37]: Yes. This is what I meant.

**Pull request closed without merging**